executor/aggfuncs/func_json_arrayagg_test.go

```diff
@@ -0,0 +1,156 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs_test
+
+import (
+	. "github.com/pingcap/check"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/executor/aggfuncs"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/types/json"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+func (s *testSuite) TestMergePartialResult4JsonArrayagg(c *C) {
+	typeList := []byte{mysql.TypeLonglong, mysql.TypeDouble, mysql.TypeString, mysql.TypeJSON}
+	var argCombines [][]byte
+	for i := 0; i < len(typeList); i++ {
+		argTypes := []byte{typeList[i]}
+		argCombines = append(argCombines, argTypes)
+	}
+
+	var tests []multiArgsAggTest
+	numRows := 5
+
+	for k := 0; k < len(argCombines); k++ {
+		entries1 := make([]interface{}, 0)
+		entries2 := make([]interface{}, 0)
+		entries3 := make([]interface{}, 0)
+
+		argTypes := argCombines[k]
+		genFunc := getDataGenFunc(types.NewFieldType(argTypes[0]))
+
+		for m := 0; m < numRows; m++ {
+			arg := genFunc(m)
+			entries1 = append(entries1, arg.GetValue())
+		}
+		// to adapt the `genSrcChk` Chunk format
+		entries1 = append(entries1, nil)
+
+		for m := 2; m < numRows; m++ {
+			arg := genFunc(m)
+			entries2 = append(entries2, arg.GetValue())
+		}
+		// to adapt the `genSrcChk` Chunk format
+		entries2 = append(entries2, nil)
+
+		entries3 = append(entries3, entries1...)
+		entries3 = append(entries3, entries2...)
+
+		aggTest := buildMultiArgsAggTester(ast.AggFuncJsonArrayagg, argTypes, mysql.TypeJSON, numRows, json.CreateBinary(entries1), json.CreateBinary(entries2), json.CreateBinary(entries3))
+
+		tests = append(tests, aggTest)
+	}
+
+	for _, test := range tests {
+		s.testMultiArgsMergePartialResult(c, test)
+	}
+}
+
+func (s *testSuite) TestJsonArrayagg(c *C) {
+	typeList := []byte{mysql.TypeLonglong, mysql.TypeDouble, mysql.TypeString, mysql.TypeJSON}
+	var argCombines [][]byte
+	for i := 0; i < len(typeList); i++ {
+		argTypes := []byte{typeList[i]}
+		argCombines = append(argCombines, argTypes)
+	}
+
+	var tests []multiArgsAggTest
+	numRows := 5
+
+	for k := 0; k < len(argCombines); k++ {
+		entries := make([]interface{}, 0)
+
+		argTypes := argCombines[k]
+		genFunc := getDataGenFunc(types.NewFieldType(argTypes[0]))
+
+		for m := 0; m < numRows; m++ {
+			arg := genFunc(m)
+			entries = append(entries, arg.GetValue())
+		}
+		// to adapt the `genSrcChk` Chunk format
+		entries = append(entries, nil)
+
+		aggTest := buildMultiArgsAggTester(ast.AggFuncJsonArrayagg, argTypes, mysql.TypeJSON, numRows, nil, json.CreateBinary(entries))
+
+		tests = append(tests, aggTest)
+	}
+
+	for _, test := range tests {
+		s.testMultiArgsAggFuncWithoutDistinct(c, test)
```

Comment:
json_arrayagg is a single arg aggfuncs, so I think we need use the test function for single args